### ISSUE TITLE
Clarify Docs on handlerRemoved

### ISSUE THREAD

**Issue by carl-mastrangelo** [netty/netty#10078]
In the latest master branch, PR https://github.com/netty/netty/pull/9970 updates channel handlers to not be able to interact with the pipeline via the removed context.  The docs for this seem to indicate that upon getting this event, the context can't be used any more:

```java
    /**
     * Gets called after the {@link ChannelHandler} was removed from the actual context and it doesn't handle events
     * anymore.
     */
    default void handlerRemoved(ChannelHandlerContext ctx) throws Exception {
        // NOOP
    }
```

However, it seems that this is not correct, as the context can be used while in the process of removal.   In `DefaultChannelPipelineTest.testRemoveAndForwardInbound`:

```java
            ChannelPipeline p = self.pipeline();
            handler1.inboundBuffer.add(8);
            assertEquals(8, handler1.inboundBuffer.peek());
            assertTrue(handler2.inboundBuffer.isEmpty());
            p.remove(handler1);
            assertEquals(1, handler2.inboundBuffer.size());
            assertEquals(8, handler2.inboundBuffer.peek());
```

Here, the call to remove triggers a channel read.  The docs indicate that this should not be the case.
In Netty 5.0, I think we should clarify what happens in handler removal, because as it stands it doesn't make sense.    Should the context passed to `handlerRemoved` point to the context that replaced it?

**Comment by normanmaurer** [comment]
@carl-mastrangelo so I understand you correct that we should update the docs to clarify the context can still be used within the method but not after the method returns ?

**Comment by carl-mastrangelo** [comment]
@normanmaurer yes.   As it stands, the docs on the method imply that the context has been removed, which is not the case.  I think it's reasonable to say that the ctx cannot be used after handlerRemoved has returned.

One other clarification: is it allowed to add a handler with the same name within a handlerRemoved() call?
